### PR TITLE
Don't Stand Under Shuttles

### PR DESCRIPTION
**Pull request by Aurorablade**
Made cause tiger brought it up.
I chose a bluespace sprite from bay, this is TGs shuttle rpple code.

:cl: Fethas
rscadd: There will now be an visual indicator to annouce that an incomming shuttle is about yo wreck you.
/:cl:

### CODE REVIEW

**Comment by taukausanake** [comment]
but what fun is that?

**Comment by LightFire53** [comment]
Check yourself before you wreck yourself.

Also check before the shuttle wrecks you. Don't get hit by a high speed pod dropping out of warp speed.

**Comment by KasparoVy** [comment]
Got a gif of this in action? I bet it looks dope.

**Comment by Aurorablade** [comment]
had trouble catching it.

**Comment by Aurorablade** [comment]
@Crazylemon64 also sprite label technically...

**Comment by jughu** [comment]
How can you stand under the shuttle if theres no floor in space? 🤔

**Comment by Markolie** [comment]
🤔

**Comment by KasparoVy** [comment]
You're still in the standing position, you just don't happen to be standing _on_ anything.

**Comment by DesolateG** [comment]
@KasparoVy are you sure you're not laying down? in space there is no up down left or right, it's all about perspective ;)

**Comment by Fox-McCloud** [comment]
Needs pics.

**Comment by KingPhilipIII** [comment]
If were adding the ripples, we should readd being turned into roadkill. Now you have warning that you'll get smashed if you don't move out of the way.

**Comment by SamHPurp** [comment]
Not a fan, less hugboxy not more!

But I understand the purpose behind this.

**Comment by Aurorablade** [comment]
again not that good at getting it. and considering its not there there the whole time...yeah...

**Comment by Aurorablade** [comment]
@KingPhilipIII ...you can already be turned into roadkill.

Adding gif sortly, note i am using a bluespace tile thing from page, rather then TGs circles. But if people do not like the bluespace i can use the circles. Please note its only goint to start giving the visual queue about 10 seconds towards dock time. For the gif i used the engineering doc as i did not want to wait for the emergency shuttle.
Also you already got a text warning before the shuttle ran you down.
This isn't going to remove the fact that the shuttle will gib/maim/make you loose body parts when it hits you..

**Comment by Spacemanspark** [comment]
:+1: I like the look.

**Comment by taukausanake** [comment]
I just thought of something. If you know where the shuttle is going to be before it is and you have escape and are waiting in space for it so you can ride in the front this makes it easier to do that. Sure 9/10 times the antag is going to get away with the shuttle but this does make that easier

**Comment by Aurorablade** [comment]
Okay i am changing to TGs sprites it nows lookes at thus.

**Comment by tigercat2000** [comment]
@taukausanake It's easily predictable exactly where a shuttle will land with an SPS and the map editor open. This is more for new players than anything, to make sure they don't get smashed by the deadly shuttle and figure out they need to gtfo.

@Aurorablade I definitely prefer Baystation's version. I do notice a little bit of an odd issue with both gifs, though; Why is the front window not putting down an overlay? It should, it has a turf under it. Also, what's with the corner lighting bugging out? Is that normal?

**Comment by Spacemanspark** [comment]
Wait why did you change it.

**Comment by Aurorablade** [comment]
@tigercat2000 theres not floor title under the window. Not sure about the corner light unless i compare to a another of my branches without this code.

@Spacemanspark i think someone said they liked tgs version..but hmmm...

**Comment by Aurorablade** [comment]
@tigercat2000 yeaaahhh...its a goonlight thing....

**Comment by Fox-McCloud** [comment]
The circles are better, IMO, because it doesn't just look as much like a floating block in space and is more reminiscent of indentations from shuttle engines creating a burn pattern.

**Comment by tigercat2000** [comment]
@Fox-McCloud They are one-color blue circles almost certainly drawn using dream maker's circle brush in less than 10 seconds. They don't look anything like burn patterns.

**Comment by Lady-Luck** [comment]
I think the squares look better but I'd be happy with either one.

**Comment by taukausanake** [comment]
Can't you make some icon that smooths with others next to it like walls? Use the former solid bluespace effect and smooth them together. Circles look dumb

**Comment by Aurorablade** [comment]
I am havle trouble smoothing it...

**Comment by taukausanake** [comment]
Trouble coding the smoothing or spriting it?

**Comment by Rumiluntti** [comment]
Bluespess things look **way** better than the blue circles.

**Comment by Aurorablade** [comment]
now if i could figure out how to smooth it right...it has TILES for smoothing..but...



@tigercat2000 you know on that?

**Comment by Fox-McCloud** [comment]
you could try `smooth = SMOOTH_TRUE`

I'm not sure if that'll work or not though.

**Comment by Aurorablade** [comment]
idid..and it IS Smoothing but in an odd way...

**Comment by Aurorablade** [comment]
cause the base sprite dosen't have those borders

**Comment by scrubmcnoob** [comment]
@Fox-McCloud @Aurorablade What's happening with this?

File: code/modules/shuttle/ripple.dm [hunk 1]
```diff
@@ -0,0 +1,22 @@
+/obj/effect/ripple
```
Comment (Fox-McCloud) [inline code/modules/shuttle/ripple.dm]: /obj/effect/overlay/temp/ripple

File: code/modules/shuttle/ripple.dm [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+	desc = "Something is coming through hyperspace, you can see the \
+		visual disturbances. It's probably best not to be on top of these \
+		when whatever is tunneling comes through."
+	icon = 'icons/turf/floors/ripple.dmi'
+	icon_state = "ripple"
+	anchored = TRUE
+	density = FALSE
+	smooth = SMOOTH_TRUE
+	layer = RIPPLE_LAYER
+	alpha = 0
```
Comment (Fox-McCloud) [inline code/modules/shuttle/ripple.dm]: if you make this a child of `temp` then you can just have the duration set to `duration = 3 * SHUTTLE_RIPPLE_TIME`

File: code/modules/shuttle/shuttle.dm [hunk 3]
```diff
@@ -329,6 +331,11 @@
 	if(mode != SHUTTLE_CALL)
 		return
 
+	if(ripples.len)
```
Comment (Fox-McCloud) [inline code/modules/shuttle/shuttle.dm]: Make this into a proc `remove_ripples`

File: code/modules/shuttle/shuttle.dm [hunk 4]
```diff
@@ -378,6 +385,29 @@
...
+		if(!T1)
+			continue
+		if(T0.type != T0.baseturf)
+			ripple_turfs += T1
+
+	return ripple_turfs
+
 //this is the main proc. It instantly moves our mobile port to stationary port S1
 //it handles all the generic behaviour, such as sanity checks, closing doors on the shuttle, stunning mobs, etc
 /obj/docking_port/mobile/proc/dock(obj/docking_port/stationary/S1, force=FALSE)
```
Comment (Fox-McCloud) [inline code/modules/shuttle/shuttle.dm]:
```
	if(S1.get_docked() == src)
		remove_ripples()
		return
```

**Comment by Fox-McCloud** [comment]
Needs a few tweaks; TG updated their ripple system a few times--would probably be best to follow suit.

File: code/modules/shuttle/ripple.dm [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+	name = "hyperspace ripple"
+	desc = "Something is coming through hyperspace, you can see the \
+		visual disturbances. It's probably best not to be on top of these \
+		when whatever is tunneling comes through."
+	icon = 'icons/turf/floors/ripple.dmi'
+	icon_state = "ripple"
+	anchored = TRUE
+	density = FALSE
+	smooth = SMOOTH_TRUE
+	layer = RIPPLE_LAYER
```
Comment (Fox-McCloud) [inline code/modules/shuttle/ripple.dm]: `mouse_opacity = 1`

File: code/modules/shuttle/shuttle.dm [hunk 6]
```diff
@@ -426,6 +463,11 @@
 	//move or squish anything in the way ship at destination
 	roadkill(L0, L1, S1.dir)
 
+	// Removes ripples
+	for(var/i in ripples)
```
Comment (Fox-McCloud) [inline code/modules/shuttle/shuttle.dm]: `remove_ripples()`

put this above roadkill, too.

File: code/modules/shuttle/shuttle.dm [hunk 7]
```diff
@@ -565,6 +607,9 @@
 //used by shuttle subsystem to check timers
 /obj/docking_port/mobile/proc/check()
 	var/timeLeft = timeLeft(1)
+	if(!ripples.len && (timeLeft <= SHUTTLE_RIPPLE_TIME) && ((mode == SHUTTLE_CALL) || (mode == SHUTTLE_RECALL)))
+		create_ripples(destination)
+
```
Comment (Fox-McCloud) [inline code/modules/shuttle/shuttle.dm]:
```
/obj/docking_port/mobile/proc/check_effects()
	if(!ripples.len)
		if((mode == SHUTTLE_CALL) || (mode == SHUTTLE_RECALL))
			var/tl = timeLeft(1)
			if(tl <= SHUTTLE_RIPPLE_TIME)
				create_ripples(destination)
```

File: code/modules/shuttle/ripple.dm [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+	alpha = 0
+	duration = 3 * SHUTTLE_RIPPLE_TIME
+
+/obj/effect/overlay/temp/ripple/New()
+	. = ..()
+	smooth_icon(src)
+	animate(src, alpha=255, time=SHUTTLE_RIPPLE_TIME)
+	// In case something goes wrong, delete us in a bit
+	addtimer(src, "delself", 3 * SHUTTLE_RIPPLE_TIME, FALSE)
```
Comment (Fox-McCloud) [inline code/modules/shuttle/ripple.dm]: no longer needed; `temp` overlays delete themselves.

File: code/modules/shuttle/ripple.dm [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+
+/obj/effect/overlay/temp/ripple/proc/delself()
```
Comment (Fox-McCloud) [inline code/modules/shuttle/ripple.dm]: no longer needed.

**Pull request merged**